BookingProject/BookingWebAPI/BookingWebApi.Application/User/Services/AppUserService.cs

```diff
@@ -12,15 +13,31 @@ namespace BookingWebApi.Application.User.Services
 {
     public interface IAppUserService
     {
-        Task<Result<UserChangeDto>> UpdateUser(string userId, UserUpdateQuery query, CancellationToken cancellationToken);
+        Task<Result<UserChangeDto>> UpdateUserAsync(string userId, UserUpdateQuery query, CancellationToken cancellationToken);
+        Task<Result<UserInfo>> GetUserInfoAsync(string userId);  
     }
     public class AppUserService(
         IAppUserRepository _appUserRepository,
         ILogger<AppUserService> _logger,
-       IUserChangeKafkaProducer _kafka
+        IUserChangeKafkaProducer _kafka,
+        IRedisCacheService _cache
     ) : IAppUserService
     {
-        public async Task<Result<UserChangeDto>> UpdateUser(string userId, UserUpdateQuery query, CancellationToken cancellationToken)
+        public async Task<Result<UserInfo>> GetUserInfoAsync(string userId)
+        {
+            var cacheUserInfo = await _cache.GetAsync<UserInfo>(userId);
+
+            if (cacheUserInfo != null) return Result<UserInfo>.Success(cacheUserInfo);
+
+            var userInfo = await _appUserRepository.GetUserInfoById(userId);
+            if (userInfo is null) return Result<UserInfo>.Failure(userInfo.ErrorMessage);
+
+            await _cache.SetAsync(userId, new UserInfo(userInfo.Value.UserName, userInfo.Value.Email), TimeSpan.FromHours(1));
```

Comment:
Not good approach to create keys like this. You better make `User_{userId}` as key or something